homeassistant/components/zwave/config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Config flow to configure Z-Wave."""
+from collections import OrderedDict
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+
+from .const import (
+    CONF_USB_STICK_PATH, CONF_NETWORK_KEY,
+    DEFAULT_CONF_USB_STICK_PATH, DOMAIN)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZwaveFlowHandler(config_entries.ConfigFlow):
+    """Handle a Z-Wave config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Z-Wave config flow."""
+        self.usb_path = CONF_USB_STICK_PATH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason='one_instance_only')
+
+        errors = {}
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_USB_STICK_PATH,
+                            default=DEFAULT_CONF_USB_STICK_PATH)] = str
+        fields[vol.Optional(CONF_NETWORK_KEY)] = str
+
+        if user_input is not None:
+            # Check if USB path is valid
+            from openzwave.option import ZWaveOption
+            from openzwave.object import ZWaveException
+
+            try:
+                # pylint: disable=unused-variable
+                option = ZWaveOption(  # noqa: F841
+                    user_input[CONF_USB_STICK_PATH],
+                    user_path=self.hass.config.config_dir
+                )
+            except ZWaveException:
```

Comment:
What kind of error can this be? Is it testing that the stick works? Because if so, it's doing I/O and thus you need to run this in the executor:

```python
from functools import partial
option = await self.hass.async_add_executor_job(partial(ZWaveOption, user_input[CONF_USB_STICK_PATH], user_path=self.hass.config.config_dir))
```

Reply:
Yeah, it's checking that the usb stick is valid. Switched to run in executor as requested.